Week4/app/src/main/java/com/em4n0101/mymovies/MoviesAdapter.kt

```diff
@@ -0,0 +1,30 @@
+package com.em4n0101.mymovies
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.em4n0101.mymovies.data.Movie
+import com.em4n0101.mymovies.utils.MovieViewHolder
+
+class MoviesAdapter(private val movieList: ArrayList<Movie>, val delegate: SelectItemListener): RecyclerView.Adapter<MovieViewHolder>() {
+
+    interface SelectItemListener {
+        fun listItemPressed(list: Movie)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {
+        val holder = LayoutInflater.from(parent.context)
+            .inflate(R.layout.movie_view_holder, parent, false)
+        return MovieViewHolder(holder)
+    }
+
+    override fun getItemCount() = movieList.size
+
+    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
+        holder.title.text = movieList[position].title
+        holder.poster.setImageResource(movieList[position].posterResource)
+        holder.itemView.setOnClickListener {
+            delegate.listItemPressed(movieList[position])
+        }
```

Comment:
I like to move this code to the Viewholder class where I would create a method called something like `bind()`